wp/lib/bap_wp/src/compare.ml

```diff
@@ -23,6 +23,14 @@ module Env = Environment
 module Pre = Precondition
 module Constr = Constraint
 
+type 'a code =
```

Comment:
Is this record only used in comparative analyses? Is there a use-case for a single analysis?

Reply:
Sorry, my confusion. This was placed in the `compare` module so I was under the assumption that we only used it for comparisons and not single analyses. It felt a bit weird to see some parts of `single` use a type from the `compare` module like

```
| Single of {
      pre : Constr.t;
      orig : sub Comp.code
    }
```